lib/galaxy/celery/tasks.py

```diff
@@ -389,3 +393,48 @@ def prune_history_audit_table(sa_session: galaxy_scoped_session):
 def cleanup_short_term_storage(storage_monitor: ShortTermStorageMonitor):
     """Cleanup short term storage."""
     storage_monitor.cleanup()
+
+
+@galaxy_task(action="clean up job working directories")
+def cleanup_jwds(sa_session: galaxy_scoped_session, object_store: BaseObjectStore):
+    """Cleanup job working directories for failed jobs that are older than X days"""
+
+    def get_failed_jobs():
+        failed_jobs = {}
+        jobs = sa_session.query(model.Job).filter(
+            model.Job.state == "error",
+            model.Job.update_time > datetime.datetime.now() - datetime.timedelta(days=days),
+            model.Job.object_store_id is not None,
+        )
+
+        for job in jobs:
+            failed_jobs[job.id] = object_store.get_filename(job, base_dir="job_work", dir_only=True, obj_dir=True)
+
+        return failed_jobs
+
+    def delete_jwd(jwd_path):
+        try:
+            # Validate that the path is a JWD
+            # It is a JWD if the following conditions are true:
+            # 1. Check if tool_script.sh exists
+            # 2. Check if directories 'inputs', and 'outputs' exist
+            if (
+                os.path.exists(jwd_path)
+                and os.path.exists(f"{jwd_path}/tool_script.sh")
+                and os.path.exists(f"{jwd_path}/inputs")
+                and os.path.exists(f"{jwd_path}/outputs")
+            ):
+                shutil.rmtree(jwd_path)
+        except OSError as e:
+            log.error(f"Error deleting job working directory: {jwd_path} : {e.strerror}")
```

Comment:
```suggestion
        try:
            # Get job working directory from object store
            path = object_store.get_filename(job, base_dir="job_work", dir_only=True, obj_dir=True)
            shutil.rmtree(jwd_path)
        except ObjectNotFound:
            # job working directory already deleted
            pass
        except OSError as e:
            log.error(f"Error deleting job working directory: {jwd_path} : {e.strerror}")
```
and `from galaxy.exceptions import ObjectNotFound` after line 26.